### PR TITLE
Add Pd::WorkshopSurvey clientside code

### PR DESCRIPTION
**Pull request by Hamms**
Also adds (and takes advantage of) the ability for the `buildSelectFieldGroupFromOptions` helper to accept either Array- or Object-defined options, and a new `getCurrentScriptData` helper for getting and serializing all the data attributes attached to the current script tag.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_survey/WorkshopSurvey.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    if (this.isLocalSummer()) {
+      return [
+        WorkshopQuality,
+        PersonalInvolvement,
+        WorkshopResults,
+        Demographics,
+      ];
+    } else {
+      return [
+        InvalidWorkshop,
```
Comment (Hamms) [inline apps/src/code-studio/pd/workshop_survey/WorkshopSurvey.jsx:49]: The plan is for this to be a temporary component, and for the old workshop survey form to be migrated over

File: apps/src/code-studio/pd/workshop_survey/Demographics.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+const LABELS = {
+  "race": "Race (check all that apply)",
+  "yearsTaught": "If you are/were a K-12 teacher, how many years have you taught?",
+  "highestEducation": "What is your highest level of education?",
+  "degreeField": "What field of study is/are your degree(s) in?",
+  "yearsTaughtStem": "For how many years have you taught a STEM subject?",
+  "yearsTaughtCs": "For how many years have you taught Computer Science?",
+  "haveRequiredLicenses": "Do you have the required licenses / certifications to teach CS in your school district?"
+};
```
Comment (aoby) [inline apps/src/code-studio/pd/workshop_survey/Demographics.jsx:13]: Where are the options for these?

File: apps/src/util/getScriptData.js [hunk 3]
```diff
@@ -25,3 +25,44 @@ export default function getScriptData(name) {
...
+ *  {
+ *    someServerObject: {
+ *      userId: "foobar"
+ *    },
+ *    someServerString: "hello"
+ *  }
+ *
+ * @returns an object containing the decoded json values of each of the data attributes
+ */
+export function getCurrentScriptData() {
```
Comment (aoby) [inline apps/src/util/getScriptData.js:52]: Nice documentation 👍

File: apps/src/code-studio/pd/workshop_survey/Demographics.jsx (hunk 2 above)
Comment (aoby) [reply apps/src/code-studio/pd/workshop_survey/Demographics.jsx:13]: Per offline discussion, these are passed in as data attributes from the server [here](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/local_summer_workshop_survey.rb#L50) (so they can be validated). 👍

**Pull request merged**